pkg/apis/elasticsearch/v1/name.go

```diff
@@ -200,3 +200,13 @@ func StackConfigAdditionalSecretName(esName string, secretName string) string {
 	secretNameHash := hash.HashObject(secretName)
 	return ESNamer.Suffix(esName, "scp", secretNameHash)
 }
+
+// PodDisruptionBudgetNameForRole returns the name of the PodDisruptionBudget for a given Elasticsearch cluster name and role.
+func PodDisruptionBudgetNameForRole(esName string, role string) string {
+	name := DefaultPodDisruptionBudget(esName) + "-" + role
+	// For coordinating nodes (no roles), append "coordinating" to the name
+	if role == "" {
+		name += "coordinating"
+	}
+	return name
```

Comment:
I think it is the purpose of the namer to add the `-`:
```suggestion
	if role == "" {
		role = "coordinating"
	}
	return ESNamer.Suffix(esName, defaultPodDisruptionBudget, role)
```